test/cmd/api.sh

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/bash
+
+#
+# Test osbuild-composer's main API endpoint by building a sample image and
+# uploading it to AWS.
+#
+# This script sets `-x` and is meant to always be run like that. This is
+# simpler than adding extensive error reporting, which would make this script
+# considerably more complex. Also, the full trace this produces is very useful
+# for the primary audience: developers of osbuild-composer looking at the log
+# from a run on a remote continuous integration system.
+#
+
+set -euxo pipefail
+
+
+#
+# Verify that this script is running in the right environment. In particular,
+# it needs variables is set to access AWS.
+#
+
+printenv AWS_REGION AWS_BUCKET AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY > /dev/null
+
+
+#
+# Create a temporary directory and ensure it gets deleted when this script
+# terminates in any way.
+#
+
+WORKDIR=$(mktemp -d)
+function cleanup() {
+  rm -rf "$WORKDIR"
+}
+trap cleanup EXIT
+
+
+#
+# Install the aws client from the upstream release, because it doesn't seem to
+# be available as a RHEL package.
+#
+
+if ! hash aws; then
+    mkdir "$WORKDIR/aws"
+    pushd "$WORKDIR/aws"
+        curl -Ls --retry 5 --output awscliv2.zip \
+            https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip
+        unzip awscliv2.zip > /dev/null
+        sudo ./aws/install > /dev/null
+        aws --version
+    popd
+fi
+
+AWS_CMD="aws --region $AWS_REGION --output json --color on"
+
+
+#
+# Prepare a request to be sent to the composer API.
+#
+
+REQUEST_FILE="${WORKDIR}/request.json"
+SNAPSHOT_NAME=$(uuidgen)
+
+cat > "$REQUEST_FILE" << EOF
+{
+  "distribution": "rhel-8",
+  "image_requests": [
+    {
+      "architecture": "x86_64",
+      "image_type": "ami",
+      "repositories": [
+        {
+          "baseurl": "http://download.eng.rdu.redhat.com/rel-eng/rhel-8/RHEL-8/RHEL-8.3.0-Beta-1.0/compose/BaseOS/x86_64/os",
```

Comment:
Just saw tests are failing: for `fedora-*` tests, you'll need to change this request to fedora ones, unless we wanna test cross-distro as well.

Reply:
They do not. Thanks for pointing it out. I've changed it to build fedora-31 (32 cannot be built on other systems currently, due to an selinux bug).